Lib/fontTools/ttLib/woff2.py

```diff
@@ -1168,26 +1168,8 @@ def __init__(self, reader=None, data=None, transformedTables=None):
 				raise ValueError(
 					"'glyf' and 'loca' must be transformed (or not) together"
 				)
-
-		self.majorVersion = None
-		self.minorVersion = None
-		self.metaData = None
-		self.privData = None
+		super(WOFF2FlavorData, self).__init__(reader=reader)
 		if reader:
-			self.majorVersion = reader.majorVersion
```

Comment:
I don't think removing this whole block is the correct thing to do here

Reply:
I think you're missing the dependency between the two things. WOFF2FlavorData has almost the exact same initializer (with some additions) as its superclass, except for the fact that the decompression method is different. If you split out the decompression part to a method (which gets overridden by WOFF2FlavorData), then you can remove a lot of repeated code by calling super.__init__.